### PR TITLE
VMware: make properties configurable

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
This allows user to control set of properties which can be included
in the hostvars.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
lib/ansible/plugins/inventory/vmware_vm_inventory.py

### CODE REVIEW

**Comment by Akasurde** [comment]
cc @angystardust I made properties configurable. This will help to improvise performance while gathering VM inventory.

**Comment by Akasurde** [comment]
@pdellaert @dericcrago @Im0 @jeking3  Could you please review this ?

**Comment by angystardust** [comment]
One more thing I would like to have in the new inventories (both for vms and esxi hosts) that was possibile in the older inventory script is the `alias_pattern` parameter to alter the default inventory object names (so, for example, one can choose to strip the uuid)
Right now I’m editing the Python file in order to do that :)

**Comment by jeking3** [comment]
Does this build on the work I did in vmware_guest_facts and pyVmomi to extract raw vSphere API properties?

**Comment by Akasurde** [comment]
> Does this build on the work I did in vmware_guest_facts and pyVmomi to extract raw vSphere API properties?

No. Since your fix is available in 6.7 and onwards. But I can include that change as well with fallback mechanism.

**Review by adarobin — APPROVED** [review]
LGTM

**Review by alongchamps — APPROVED** [review]
LGTM

**Comment by jeking3** [comment]
> > Does this build on the work I did in vmware_guest_facts and pyVmomi to extract raw vSphere API properties?
> 
> No. Since your fix is available in 6.7 and onwards. But I can include that change as well with fallback mechanism.

Just to clarify, you need `pyvmomi>6.7.1` to get the pyVmomi json conversion of vSphere objects.

File: lib/ansible/plugins/inventory/vmware_vm_inventory.py [hunk 1]
```diff
@@ -61,6 +61,19 @@
             - 'https://code.vmware.com/web/sdk/60/vcloudsuite-python'
             default: False
             type: boolean
+        properties:
+            description:
+            - Specify the list of Property associated with VM.
```
Comment (jeking3) [inline lib/ansible/plugins/inventory/vmware_vm_inventory.py]:
```suggestion
            - Specify the list of VMware schema properties associated with the VM.
```

File: lib/ansible/plugins/inventory/vmware_vm_inventory.py [hunk 2]
```diff
@@ -61,6 +61,19 @@
...
+            - These properties will be populated in hostvars of the given VM.
+            - Each value in list specifies the path to a specific property in VM object.
```
Comment (jeking3) [inline lib/ansible/plugins/inventory/vmware_vm_inventory.py]:
```suggestion
            - Each value in the list specifies the path to a specific property in VM object.
```

File: lib/ansible/plugins/inventory/vmware_vm_inventory.py [hunk 3]
```diff
@@ -61,6 +61,19 @@
...
+            type: list
+            default: [ 'name', 'config.cpuHotAddEnabled', 'config.cpuHotRemoveEnabled',
```
Comment (jeking3) [inline lib/ansible/plugins/inventory/vmware_vm_inventory.py:69]: tags?

Does it make sense to have a default list of properties?

**Review by jeking3 — CHANGES_REQUESTED** [review]
I haven't worked on an inventory module before, but I'm wondering why the list of default properties is what it is.

**Pull request merged**